rts-front/src/main.rs

```diff
@@ -0,0 +1,156 @@
+mod contexts;
+mod routes;
+mod screens;
+mod utils;
+
+use reqwasm::http::Request;
+use yew::prelude::*;
+use yew_router::prelude::*;
+
+use crate::contexts::*;
+use crate::routes::Route;
+use crate::screens::ai_upload::AIUpload;
+use crate::screens::login::Login;
+use crate::screens::register::Registration;
+use crate::utils::alert_message;
+
+#[function_component(HomeScreen)]
+fn home_screen() -> Html {
+    html! {
+        <h1>{ "Home screen!" }</h1>
+    }
+}
+
+#[function_component(Leaderboard)]
+fn leaderboard() -> Html {
+    html! {
+        <h1>{ "Leaderboard!" }</h1>
+    }
+}
+
+#[function_component(PageNotFound)]
+fn page_not_found() -> Html {
+    html! {
+        <h1>{ "Page not found" }</h1>
+    }
+}
+
+#[function_component(NavigationBar)]
+fn navigation_bar() -> Html {
+    let login_context = use_context::<LoginContext>().expect("no context found");
+    let history = use_history().unwrap();
+
+    let on_logout_click = {
+        let login_context = login_context.clone();
```

Comment:
Can't we use a smart pointer instead of cloning the whole context ? Like `Rc` or `RefCell`. See docs: https://yew.rs/docs/advanced-topics/optimizations#using-smart-pointers-effectively

Reply:
The context is a ReducerHandle, which already uses a Rc
https://docs.rs/yew/latest/src/yew/functional/hooks/use_reducer.rs.html#34

I don't think it's necessary to add our own Rc on top?